Update id_assoc column in the job table.

chirag06 opened the pull request:
Updated the script and readme file.

Commit:
Modified code to add the missing id_assoc in the job table

1) This fix will allow sacct command to display username and account name.

Commit:
Updated slurmdb-fix-for-xdmod-unk.md readme file to fix the id_assoc column in the job table.

eesaanatluri commented on slurmdb-fix-job-null-accounts.sh:
```diff
@@ -2,17 +2,12 @@
-
-	#mysql -u root -D slurmdb -e \"UPDATE ohpc_job_table SET account=$slurm_acct_name WHERE id_user=$uid and account IS NULL;\"
+	slurm_user_name=`echo $line | cut -d ',' -f2`
+	data1=$(mysql -u root -D slurmdb<<<"SELECT distinct(id_assoc) , acct, user  FROM ohpc_assoc_table  where user='$slurm_user_name';")
```
Could you change the var name to `slurm_assocs` instead of `data1` on this line.
Also, put your sql query inside the `` rather than including them inside $() and change the <<< to use -e 
These changes are suggested to make the code readable to the user.

eesaanatluri commented on slurmdb-fix-job-null-accounts.sh:
```diff
@@ -2,17 +2,12 @@
+	data1=$(mysql -u root -D slurmdb<<<"SELECT distinct(id_assoc) , acct, user  FROM ohpc_assoc_table  where user='$slurm_user_name';")
+	id_assoc=`echo $data1 | cut -d ' ' -f4`
+	acct=`echo $data1 | cut -d ' ' -f5`
+	user=`echo $data1 | cut -d ' ' -f6`
```
From your select query above both `user` and `slurm_user_name` variables are same if you have to use it then you can directly refer to `slurm_user_name` , I think it is redundant. Also, you do not seem to use it in your update query later . I think you can remove this line.

eesaanatluri commented on slurmdb-fix-job-null-accounts.sh:
```diff
@@ -2,17 +2,12 @@
+	id_assoc=`echo $data1 | cut -d ' ' -f4`
+	acct=`echo $data1 | cut -d ' ' -f5`
+	user=`echo $data1 | cut -d ' ' -f6`
+	data1=$(mysql -u root -D slurmdb<<<"UPDATE ohpc_job_table  set id_assoc='$id_assoc' where id_user='$uid' and id_assoc=0;\
```
Please print the update statements you intend to apply like I did on line 15 in the org file(you may have two update statements), it's a good practice to do it this way because you can have the exact number of update statements that you can match to the number of problematic records you observed from your select statement.

eesaanatluri replied:
While applying this patch in the prod you can write the update stmts to a file then apply all of them at once, this can improve confidence by giving us a chance to do sanity checks before hand rather than after the fact.

Commit:
Applied following changes for better Readability of code

1) Changed variable data1 to slurm_assocs.
2) sql query inside the `` rather than including them inside $() and changed the <<< to use -e in the command.
3) removed user variable as it was not used in the code.
4) Added progress percent tracker.
5) Added print staments for sanity checking and debugging if required.
